sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,213 @@
+     *
+     * @Param senderAsyncClient Service Bus Sender Async Client
+     * @Param maxMessages Maximum Number Of Messages
+     */
+    void sendMessagesAsync(ServiceBusSenderAsyncClient senderAsyncClient, int maxMessages) {
+        List<HashMap<String, String>> data =
+            GSON.fromJson(
+                "[" +
+                    "{'name' = 'Einstein', 'firstName' = 'Albert'}," +
+                    "{'name' = 'Heisenberg', 'firstName' = 'Werner'}," +
+                    "{'name' = 'Curie', 'firstName' = 'Marie'}," +
+                    "{'name' = 'Hawking', 'firstName' = 'Steven'}," +
+                    "{'name' = 'Newton', 'firstName' = 'Isaac'}," +
+                    "{'name' = 'Bohr', 'firstName' = 'Niels'}," +
+                    "{'name' = 'Faraday', 'firstName' = 'Michael'}," +
+                    "{'name' = 'Galilei', 'firstName' = 'Galileo'}," +
+                    "{'name' = 'Kepler', 'firstName' = 'Johannes'}," +
+                    "{'name' = 'Kopernikus', 'firstName' = 'Nikolaus'}" +
+                    "]",
+                new TypeToken<List<HashMap<String, String>>>() {
+                }.getType());
+
+        for (int i = 0; i < Math.min(data.size(), maxMessages); i++) {
+            final String messageId = Integer.toString(i);
+            ServiceBusMessage message = new ServiceBusMessage(GSON.toJson(data.get(i), Map.class).getBytes(UTF_8));
+            message.setContentType("application/json");
+            message.setSubject(i % 2 == 0 ? "Scientist" : "Physicist");
+            message.setMessageId(messageId);
+            message.setTimeToLive(Duration.ofMinutes(2));
+            System.out.printf("Message sending: Id = %s\n", message.getMessageId());
+            senderAsyncClient.sendMessage(message)
+                .doOnSuccess(onSuccess -> System.out.printf("\tMessage acknowledged: Id = %s\n", message.getMessageId()))
+                .block();
+        }
+    }
+
+    /**
+     * Receive {@link ServiceBusMessage messages} and dead letter it within an Azure Service Bus Queue
+     *
+     * @Param connectionString Service Bus Connection String
+     * @Param queueName Queue Name
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    void exceedMaxDelivery(String connectionString, String queueName) throws InterruptedException {
+        ServiceBusReceiverAsyncClient receiverAsyncClient
+            = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .receiver()
+            .queueName(queueName)
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .buildAsyncClient();
+        receiverAsyncClient.receiveMessages().subscribe(receiveMessage -> {
+            System.out.printf("Picked up message; DeliveryCount %d\n", receiveMessage.getDeliveryCount());
+            receiverAsyncClient.abandon(receiveMessage);
```

Comment:
Suggest add a comment like "// return message back to the queue". I though `abandon` was `deadletter` when the first time I saw it.

Reply:
According to your comment, fixed in new version.